tests/plugins/test_default_qubit_tf.py

```diff
@@ -1236,3 +1237,55 @@ def cost(weights):
 
         assert isinstance(grad, tf.Tensor)
         assert grad.shape == weights.shape
+
+
+class TestWiresIntegration:
+    """Test that the device integrates with PennyLane's wire management."""
+
+    def make_circuit_prob(self, wires):
+        """Factory for a qnode returning probabilities using arbitrary wire labels."""
+        dev = qml.device("default.qubit.tf", wires=wires)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.RX(0.5, wires=wires[0])
+            qml.RY(2., wires=wires[1])
+            qml.CNOT(wires=[wires[0], wires[2]])
+            qml.CNOT(wires=[wires[2], wires[1]])
+            return qml.probs(wires=wires)
+
+        return circuit
+
+    def make_circuit_expval(self, wires):
+        """Factory for a qnode returning expvals using arbitrary wire labels."""
+        dev = qml.device("default.qubit.tf", wires=wires)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.RX(0.5, wires=wires[0])
+            qml.RY(2., wires=wires[1])
+            qml.CNOT(wires=[wires[0], wires[2]])
+            qml.CNOT(wires=[wires[2], wires[1]])
+            return [qml.expval(qml.PauliZ(wires=w)) for w in wires]
+
+        return circuit
+
+    @pytest.mark.parametrize("wires1, wires2", [(['a', 'c', 'd'], [2, 3, 0]),
+                                                ([-1, -2, -3], ['q1', 'ancilla', 2])])
+    def test_wires_probs(self, wires1, wires2, tol):
+        """Test that the probability vector of a circuit is independent from the wire labels used."""
+
+        circuit1 = self.make_circuit_prob(wires1)
+        circuit2 = self.make_circuit_prob(wires2)
+
+        assert np.allclose(circuit1(), circuit2(), tol)
+
+    @pytest.mark.parametrize("wires1, wires2", [(['a', 'c', 'd'], [2, 3, 0]),
+                                                ([-1, -2, -3], ['q1', 'ancilla', 2])])
+    def test_wires_expval(self, wires1, wires2, tol):
+        """Test that the expectation of a circuit is independent from the wire labels used."""
+
+        circuit1 = self.make_circuit_expval(wires1)
+        circuit2 = self.make_circuit_expval(wires2)
+
+        assert np.allclose(circuit1(), circuit2(), tol)

```

Comment:
Is it too late to think of a way of having a common place for these tests?

Reply:
Well, we kind of have...the shared device test suite in ``plugins/tests``. But it does not get run when you ``make test``, so I thought it's better to put the integration tests into every device for now.

@josh146, do you remember if we were planning to run the shared device tests automatically at some stage? They have a lot of flaky usage in them though!